hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingReadHandler.java

```diff
@@ -145,14 +145,17 @@ public void requestMigration(NonBlockingIOThread newOwner) {
 
     @Override
     public void handle() {
-        eventCount.inc();
-        lastReadTime = Clock.currentTimeMillis();
-        if (!connection.isAlive()) {
-            String message = "We are being asked to read, but connection is not live so we won't";
-            logger.finest(message);
-            return;
-        }
         try {
+            eventCount.inc();
+            // we are going to set the timestamp even if the socketChannel is going to fail reading. In that case
+            // the connection is going to be closed anyway.
+            lastReadTime = Clock.currentTimeMillis();
```

Comment:
I want to get rid of a lot of stuff in the loop :) Try catch already is available and io thread, so I'll let it take care completely of the exception handling and allow for a callback in case of an exception (close connection). If connection alive can also be removed since socket channel is closed; so you will run into that anyway.